modules/integration/src/test/scala/scala/cli/integration/CompileTestDefinitions.scala

```diff
@@ -87,7 +88,17 @@ abstract class CompileTestDefinitions(val scalaVersionOpt: Option[String])
       )
     )
     inputs.fromRoot { root =>
-      os.proc(TestUtil.cli, "compile", extraOptions, ".").call(cwd = root)
+      val output =
+        os.proc(TestUtil.cli, "compile", "--test", "--class-path", extraOptions, ".").call(cwd =
+          root).out.text().trim
+      val escapeBackslash = if (Properties.isWin) "\\\\/" else "\\/"
+      val testPathRegex =
+        s":$root/${Constants.workspaceDirName}/project_.*?/classes/test".replace(
+          "/",
+          escapeBackslash
+        ).r
+      val isDefinedTestPathInClassPath = testPathRegex.findFirstMatchIn(output).isDefined
```

Comment:
Split the lines of `output` (`output.linesIterator`), and look for one starting with `(root / Constants.workspaceDirName).toString` and ending with Seq("classes", "test").mkString(File.separator), rather than using a regex?

Reply:
Yes, good point. So I will avoid using regex.